File: datetimepickeredittext/src/main/java/io/blackbox_vision/datetimepickeredittext/internal/utils/DateUtils.java
package io.blackbox_vision.datetimepickeredittext.internal.utils;


import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.text.format.DateFormat;
import android.util.Log;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;


public final class DateUtils {
    private static final String LOG_TAG = DateUtils.class.getSimpleName();

    private static final String DATE_TEMPLATE = "dd/MM/yyyy";
    private static final String TIME_TEMPLATE = "kk:mm";

    private DateUtils() { }

    private static CharSequence format(@NonNull final Date date, @NonNull final String template) {
        return DateFormat.format(template, date);
    }

    public static CharSequence toDate(@NonNull final Date date, @Nullable final String template) {
        return format(date, null != template ? template : DATE_TEMPLATE);
    }

    public static CharSequence toTime(@NonNull final Date date, @Nullable final String template) {
        return format(date, null != template ? template : TIME_TEMPLATE);
    }

    public static Date parse(@Nullable final String dateString, @Nullable final String template) {
        final SimpleDateFormat sdf = new SimpleDateFormat(null != template ? template : DATE_TEMPLATE, Locale.getDefault());
        Date date = null;

        try {
            date = sdf.parse(dateString);
        } catch (ParseException ex) {
            Log.e(LOG_TAG, ex.getMessage(), ex.getCause());
        }

        return date;
    }
}

File: datetimepickeredittext/src/main/java/io/blackbox_vision/datetimepickeredittext/internal/utils/DateUtils.kt
package io.blackbox_vision.datetimepickeredittext.internal.utils


import android.text.format.DateFormat
import android.util.Log

import java.text.ParseException
import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale


object DateUtils {
    private val LOG_TAG = "DateUtils"

    private val DATE_TEMPLATE = "dd/MM/yyyy"
    private val TIME_TEMPLATE = "kk:mm"

    private fun format(date: Date, template: String): CharSequence {
        return DateFormat.format(template, date)
    }

    fun toDate(date: Date, template: String?): CharSequence {
        return format(date, template ?: DATE_TEMPLATE)
    }

    fun toTime(date: Date, template: String?): CharSequence {
        return format(date, template ?: TIME_TEMPLATE)
    }

    fun parse(dateString: String?, template: String?): Date? {
        val sdf = SimpleDateFormat(template ?: DATE_TEMPLATE, Locale.getDefault())
        var date: Date? = null

        try {
            date = sdf.parse(dateString)
        } catch (ex: ParseException) {
            Log.e(LOG_TAG, ex.message, ex.cause)
        }

        return date
    }

    private fun getCurrentDateTime(): String {
        return SimpleDateFormat(DATE_TEMPLATE).format(System.currentTimeMillis())
    }

    fun gettingDate(strDate: String): String {
        return strDate?.takeIf { !it.toUpperCase().equals("CR_DATE") } ?: getCurrentDateTime()
    }
}
